formal/resources/resource_connector.go

```diff
@@ -0,0 +1,163 @@
+		Description:   "Registering a Connector with Formal.",
+		CreateContext: resourceConnectorCreate,
+		ReadContext:   resourceConnectorRead,
+		UpdateContext: resourceConnectorUpdate,
+		DeleteContext: resourceConnectorDelete,
+		SchemaVersion: 1,
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(25 * time.Minute),
+		},
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Schema: map[string]*schema.Schema{
+			"id": {
+				// This description is used by the documentation generator and the language server.
+				Description: "The ID of this connector.",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			"name": {
+				// This description is used by the documentation generator and the language server.
+				Description: "Friendly name for this connector.",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"termination_protection": {
+				// This description is used by the documentation generator and the language server.
+				Description: "If set to true, this connector cannot be deleted.",
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     false,
+			},
+		},
+	}
+}
+
+func resourceConnectorCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	// use the meta value to retrieve your client from the provider configure method
+	c := meta.(*clients.Clients)
+
+	// Warning or errors can be collected in a slice type
+	var diags diag.Diagnostics
+
+	sidecarReq := &corev1.CreateConnectorRequest{
+		Name:                  d.Get("name").(string),
+		TerminationProtection: d.Get("termination_protection").(bool),
+	}
+
+	res, err := c.Grpc.Sdk.ConnectorServiceClient.CreateConnector(ctx, connect.NewRequest(sidecarReq))
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	d.SetId(res.Msg.Connector.Id)
+
+	resourceConnectorRead(ctx, d, meta)
+
+	return diags
+}
+
+func resourceConnectorRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	// use the meta value to retrieve your client from the provider configure method
+	c := meta.(*clients.Clients)
+
+	var diags diag.Diagnostics
+
+	connectorId := d.Id()
+
+	res, err := c.Grpc.Sdk.ConnectorServiceClient.GetConnector(ctx, connect.NewRequest(&corev1.GetConnectorRequest{Id: connectorId}))
+	if err != nil {
+		if connect.CodeOf(err) == connect.CodeNotFound {
+			tflog.Warn(ctx, "The Connector was not found, which means it may have been deleted without using this Terraform config.", map[string]interface{}{"err": err})
+			d.SetId("")
+			return diags
+		}
+		return diag.FromErr(err)
+	}
+
+	resApiKey, err := c.Grpc.Sdk.ConnectorServiceClient.GetConnectorApiKey(ctx, connect.NewRequest(&corev1.GetConnectorApiKeyRequest{Id: connectorId}))
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	d.Set("id", res.Msg.Connector.Id)
+	d.Set("name", res.Msg.Connector.Name)
+	d.Set("api_key", resApiKey.Msg.Secret)
```

Comment:
I believe this does not exist in the scema, is it a leftover?

Reply:
fixed 👌

Reply:
it should exist, a connector has an api key